File: 12356/army.c
#include <stdio.h>

int left[100005];
int right[100005];

int main(int argc, char *argv[])
{
	int b, s, l, r, i;


	while(1)
	{
		scanf("%d %d", &s, &b);

		if (s == 0 && b == 0)
			break;

		for(i=1; i<=s; i++) {
			left[i] = i-1;
			right[i] = i+1;
		}

		left[1] = -1;
		right[s] = -1;

		for(i=0; i<b; i++)
		{

			scanf("%d %d", &l, &r);

			left[right[r]] = left[l];
			if(left[l] != -1)
				printf("%d ", left[l]);
			else
				printf("* ");

			right[left[l]] = right[r];
			if(right[r] != -1)
				printf("%d\n", right[r]);
			else
				printf("*\n");
		}

		printf("-\n");
	}

	return 0;
}

File: 12356/army.py
import sys

while True:
	line = sys.stdin.readline().strip().split()

	s = int(line[0])
	b = int(line[1])

	if s == 0 and b == 0:
		break

	left = [-2]
	right = [-2]

	for i in range(1, s+1):
		left.append(i-1)
		right.append(i+1)

	left[1] = -1
	right[s] = -1

	for i in range(b):
		line = sys.stdin.readline().strip().split()

		l = int(line[0])
		r = int(line[1])

		x = right[r]
		y = left[l]

		# print("x,y ", x, y)

		left[right[r]] = left[l]
		if left[l] == -1:
			left_res = "*"
		else:
			left_res = str(left[l])

		right[left[l]] = right[r]
		if right[r] == -1:
			right_res = "*"
		else:
			right_res = str(right[r])


		# print(left)
		# print(right)

		res = "%s %s"%(left_res, right_res)

		print(res)

	print("-")